### PR TITLE
Fixing/reworking explosive interactions with organ manip and fishing

### LINKED ISSUE #88227: Exploding someone during organ manipulation summons new organs.
### Client Version:

515.1641

### Issue Summary:

When an explosion goes off by someone who is currently opened for organ manipulation, randomized organs will spawn from nowhere. This appears to grant easy access to implants, mutant organs, and more with nothing more than a morgue corpse and some water-potassium grenades.

image

Caused by the interaction between #87774 and the effect explosions have on fishing spots.

### Round ID:

_No response_

### Test Merge Information:

_No response_

### Reproduction Steps:

1. Open a human up for organ manipulation, up to the step where you can actually manipulate organs.
2. Set off your favorite explosive next to or on them.
3. Behold as several random organs spontaneously congeal into being, while your victim's organs stay put.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I have grown to dislike the previous implement of explosive fishing. It was particurly broken around fish sources that checked if the fishing spot was of a specific type, and the FISH_SOURCE_FLAG_EXPLOSIVE_MALUS flag only really worked around turfs because of the gap between pooling the spawn locations and spawning the loot, which would have made trying it on movables quite unsafe since they're moved to nullspace when deleted.

This has lead me to rework how explosive fishing malus/exhaustion accumulates in a simplier and more effiient way. Fish are now spawned immediately, and the counter for the malus goes up by one each call, and decays over time at a rate of 1 every 5.5 seconds.

This also fixes #88227.

## Why It's Good For The Game
Easier implementation of explosive fishing and bugfix.

## Changelog

:cl:
fix: Bombing someone during organ manipulation no longer summons new organs.
/:cl:

### CODE REVIEW

**Comment by Dawnseer** [comment]
When will we be able to sync our fishing portal to someone's chest if we multitool em during organ manipulation surgery?
Lizards gotta eat

**Comment by Ghommie** [comment]
> When will we be able to sync our fishing portal to someone's chest if we multitool em during organ manipulation surgery? Lizards gotta eat

I had thought that was already possible. If it is not, it's probably due to the different proc-chains for attacking mobs, and I'll get to it soon.

File: code/modules/fishing/sources/_fish_source.dm [hunk 1]
```diff
@@ -474,29 +475,27 @@ GLOBAL_LIST_INIT(specific_fish_icons, generate_specific_fish_icons())
...
-/datum/fish_source/proc/explosive_spawn(atom/location, severity, multiplier = 1)
+	SIGNAL_HANDLER
+	var/multiplier = 1
+	if(fish_source_flags & FISH_SOURCE_FLAG_EXPLOSIVE_MALUS)
+		if(explosive_fishing_score <= 0)
+			explosive_fishing_score = 1
+			START_PROCESSING(SSprocessing, src)
+		else
+			explosive_fishing_score++
+			multiplier = 1/(explosive_fishing_score**EXPLOSIVE_FISHING_MALUS_EXPONENT)
```
Comment (Time-Green) [inline code/modules/fishing/sources/_fish_source.dm:492]:
```suggestion
			multiplier = explosive_fishing_score**-EXPLOSIVE_FISHING_MALUS_EXPONENT
```
Unnecessary division

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/fishing/sources/_fish_source.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/fishing/sources/_fish_source.dm:492]: Aaaah, goofy ass forgot about math properties moment.

**Pull request merged**